### DISCUSSION TITLE
Creating custom torchmetrics

### DISCUSSION THREAD

**Discussion by ooghry** [autogluon/autogluon#2117] [Q&A]
Hello,
I have this custom metric:
```python
from autogluon.core.metrics import make_scorer
from sklearn.metrics import confusion_matrix
import numpy as np

def my_metric(y_true: np.ndarray, y_pred: np.ndarray, sample_weight=None) -> float:
  confusion = confusion_matrix(list(y_true),y_pred)

  return ((confusion[0][0] + confusion[1][1])*2) - (confusion[0][1] + confusion[1][0] + confusion[2][0] + confusion[2][1])

ag_scorer = make_scorer(name='myMetric_function',
  score_func=my_metric,
  optimum=1000000,
  greater_is_better=True)
```
`AG_AUTOMM` can not use any of `ag_scorer` or  `f1_macro`, give me this error:
```
/usr/local/lib/python3.7/dist-packages/autogluon/multimodal/utils.py:142: 
UserWarning: Currently, we cannot convert the metric: f1_macro to a metric supported in torchmetrics. Thus, we will fall-back to use accuracy for multi-class classification problems , ROC-AUC for binary classification problem, and RMSE for regression problems.
```
Any guide for creating a custom metric for `AG_AUTOMM` would be appreciated.

**Comment by sxjscience** [comment]
Thanks for the question! Is it possible to move it to issues so I can track it? In addition, for custom metric, what type of custom metrics are you willing to use? Internally, we rely on torchmetrics https://torchmetrics.readthedocs.io/en/stable/ to implement the metric in AutoMM.
